samples/wifi/shell/src/supp_shell.c

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* @file
+ * @brief WPA supplicant shell module
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <init.h>
+#include <ctype.h>
+
+#include <net/net_if.h>
+#include <net/wifi_mgmt.h>
+#include <net/net_event.h>
+#include "common.h"
+#include "zephyr_fmac_main.h"
+#ifdef CONFIG_WPA_SUPP
+#include "utils/common.h"
+#include "config.h"
+#include "wpa_supplicant_i.h"
+#endif /* CONFIG_WPA_SUPP */
+
+#define SUPPLICANT_SHELL_MODULE "wpa_cli"
+
+static struct {
+	const struct shell *shell;
+
+	union {
+		struct {
+			uint8_t connecting : 1;
+			uint8_t disconnecting : 1;
+			uint8_t _unused : 6;
+		};
+		uint8_t all;
+	};
+} context;
+
+static uint32_t scan_result;
+
+int cli_main(int argc, const char **argv);
+
+extern struct wpa_supplicant *wpa_s_0;
+
+struct wpa_ssid *ssid_0;
+
+#define MAX_SSID_LEN 32
+
+
+static void scan_result_cb(struct net_if *iface,
+			   int status,
+			   struct wifi_scan_result *entry)
+{
+	if (!iface) {
+		return;
+	}
+
+	if (!entry) {
+		if (status) {
+			shell_fprintf(context.shell,
+				      SHELL_WARNING,
+				      "Scan request failed (%d)\n",
+				      status);
+		} else {
+			shell_fprintf(context.shell,
+				      SHELL_NORMAL,
+				      "Scan request done\n");
+		}
+
+		return;
+	}
+
+	scan_result++;
+
+	if (scan_result == 1U) {
+		shell_fprintf(context.shell,
+			      SHELL_NORMAL,
+			      "\n%-4s | %-32s %-5s | %-4s | %-4s | %-5s\n", "Num", "SSID",
+			      "(len)", "Chan", "RSSI", "Sec");
+	}
+
+	shell_fprintf(context.shell,
+		      SHELL_NORMAL,
+		      "%-4d | %-32s %-5u | %-4u | %-4d | %-5s\n",
+		      scan_result,
+		      entry->ssid,
+		      entry->ssid_length,
+		      entry->channel,
+		      entry->rssi,
+		      (entry->security == WIFI_SECURITY_TYPE_PSK ? "WPA/WPA2" : "Open"));
+}
+
+
+static int cmd_supplicant_scan(const struct shell *shell,
+			       size_t argc,
+			       const char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_nrf_dev_ops *dev_ops = dev->api;
+
+	context.shell = shell;
+
+	return dev_ops->off_api.disp_scan(dev,
+					  scan_result_cb);
+}
+
+
+#ifdef CONFIG_WPA_SUPP
```

Comment:
This module is "WPA supplicant shell module", so I would expect that it is not included at all if CONFIG_WPA_SUPP is disabled. But it looks that only part of the commands is not available if CONFIG_WPA_SUPP is disabled.

I guess it's actually not supplicant shell, but Wi-Fi shell module, and it includes some commands which require only the wifi driver (scanning), and other commands that require supplicant.

I think this module should be renamed, or refactored in two separated modules.

Reply:
Only the `scan` command is going through the driver, so, we don't need another module only for that. We will rename it accordingly.